src/SGFTree.cpp

```diff
@@ -136,7 +136,7 @@ void SGFTree::populate_states(void) {
         std::istringstream strm(size);
         int bsize;
         strm >> bsize;
-        if (bsize <= FastBoard::MAXBOARDSIZE) {
+        if (bsize <= BOARD_SIZE) {
             // Assume 7.5 komi if not specified
             m_state.init_game(bsize, 7.5f);
             valid_size = true;
```

Comment:
line 157 has a 19 on it that should be replaced

Reply:
I'd think so, yes.

Reply:
> I don't think a 19x19 net would handle correctly a 9x9 board.

Indeed it would not due to the policy heads etc.

>if we haven't found either GM or SZ, it makes sense as a last effort to assume it's the right size

No, you should assume it's 19 because that is what the SGF spec says. If that's not BOARD_SIZE, give up.

>So in that case should we reject it if not specified and BOARD_SIZE != 19 to be consistent and follow SGF spec?

Exactly.